internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -116,6 +116,30 @@ func convertContainerHealthCheck(hc manifest.ContainerHealthCheck) *template.Con
 	}
 }
 
+func convertHostedZone(m manifest.RoutingRuleConfiguration) template.HostedZoneToAliases {
+	hostedZoneToAliases := make(map[string][]string)
+	defaultHostedZone := m.HostedZone
+	for _, alias := range m.Alias.AdvancedAliases {
+		if alias.HostedZone != nil {
+			hostedZoneToAliases[*alias.HostedZone] = append(hostedZoneToAliases[*alias.HostedZone], *alias.Alias)
+			continue
+		}
+		if defaultHostedZone != nil {
+			hostedZoneToAliases[*defaultHostedZone] = append(hostedZoneToAliases[*defaultHostedZone], *alias.Alias)
+		}
+	}
```

Comment:
Can we return `hostedZoneToAliases` already at this point? Can we assume `AdvancedAliases` and `StringSliceOrString` won't be non-empty at the same time?